test/functional/tool_wallet.py

```diff
@@ -213,11 +215,25 @@ def test_getwalletinfo_on_different_wallet(self):
     def test_salvage(self):
         # TODO: Check salvage actually salvages and doesn't break things. https://github.com/bitcoin/bitcoin/issues/7463
         self.log.info('Check salvage')
-        self.start_node(0, ['-wallet=salvage'])
+        self.start_node(0, ['-nowallet', '-wallet=salvage'])
+        address_gen_rand = random.Random(123)
+
+        def gna():
+            return self.nodes[0].getnewaddress(address_type=address_gen_rand.choice(['legacy', 'p2sh-segwit', 'bech32']))
+
+        for _ in range(101):
+            # Generate to fresh address each time
+            self.nodes[0].generatetoaddress(1, gna())
+        self.nodes[0].sendmany(amounts={gna(): 0.05 for _ in range(50)})
+        self.nodes[0].sendmany(amounts={gna(): 0.05 for _ in range(50)})
+        balances_before = self.nodes[0].getbalances()
         self.stop_node(0)
 
         self.assert_tool_output('', '-wallet=salvage', 'salvage')
```

Comment:
In commit "test: Add salvage test for wallet tool" (fa6154b3aa904eb39e6e537ff99106b6e022701b)


It seems the test is failing on this line due to stderr output https://travis-ci.org/github/bitcoin/bitcoin/jobs/735969416#L2582:

```
Salvage: WARNING: Number of keys in data does not match number of values.
Salvage: WARNING: Unexpected end of file while reading salvage output.
WARNING: WalletBatch::Recover skipping key: CDataStream::read(): end of data: unspecified iostream_category error
WARNING: WalletBatch::Recover skipping key: CDataStream::read(): end of data: unspecified iostream_category error
WARNING: WalletBatch::Recover skipping key: Error reading wallet database: CPubKey/CPrivKey corrupt
```

Reply:
#20151